aws-stackreference-architecture/application/src/application.ts

```diff
@@ -0,0 +1,169 @@
+        }, { parent: this });
+
+        this.applicationListener = new awsx.elasticloadbalancingv2.ApplicationListener(`${name}-service-alb-listener`, {
+            vpc: vpc,
+            loadBalancer: this.applicationLoadBalancer,
+            port: args.appPort,
+        }, { parent: this.applicationLoadBalancer });
+
+        this.cluster = new awsx.ecs.Cluster(`${name}-cluster`, {
+            vpc: vpc,
+            /**
+             * Prevent default security groups with `[]`. Instead
+             * provide security groups to the service directly.
+             */
+            securityGroups: [],
+            tags: {
+                ...args.baseTags,
+                Name: `${args.description} Cluster`,
+            },
+        }, { parent: vpc });
+
+        // Use the provided pre-existing security group or create a new one.
+        const appSecGroup = args.appSecurityGroupIds || [
+            new awsx.ec2.SecurityGroup(`${name}-service-sg`, {
+                vpc: vpc,
+                ingress: [
+                    {
+                        fromPort: args.appPort,
+                        toPort: args.appPort,
+                        protocol: "tcp",
+                        cidrBlocks: ["0.0.0.0/0"],
+                    },
+                ],
+                egress: [{ fromPort: 0, toPort: 0, protocol: "-1", cidrBlocks: ["0.0.0.0/0"] }],
+            }, { parent: vpc }),
+        ];
+
+        this.fargateService = new awsx.ecs.FargateService(`${name}-service`, {
+            cluster: this.cluster,
+            assignPublicIp: false,
+            subnets: args.appSubnetIds,
+            securityGroups: appSecGroup,
+            desiredCount: (args.appResources || {}).desiredCount,
+            taskDefinitionArgs: {
+                containers: {
+                    [name]: {
+                        ...args.appResources, // cpu, memory, etc.
+                        image: args.appImage,
+                        portMappings: [this.applicationListener],
+                        environment: [
+                            {
+                                name: "DB_HOST",
+                                value: args.dbHost,
+                            },
+                            {
+                                name: "DB_USERNAME",
+                                value: args.dbUsername,
+                            },
+                            {
+                                name: "DB_PASSWORD",
+                                value: args.dbPassword,
+                            },
+                            {
+                                name: "DB_PORT",
+                                value: args.dbPort,
+                            },
+                            {
+                                name: "DB_NAME",
+                                value: args.dbName,
+                            },
+                        ],
+                    },
+                },
+                // tags: {
```

Comment:
This seems like it should either be uncommented or removed?

Reply:
Fixed!